### ISSUE TITLE
[bug]: Typing a dialog request does not reflect properly in the generated files

### ISSUE THREAD

**Issue by Chappie74** [Stacked-Org/stacked#1032]
### Describe the bug

When creating a dialog, you have the ability to type the `DialogRequest` and `DialogResponse`. 
![image](https://github.com/Stacked-Org/stacked/assets/21108487/02bd3a20-e57d-4828-b374-5ef14d3078f2)

However, with a custom type, the generated file for the dialog setup does not reflect this. 
It defaults to a dynamic type. 

![image](https://github.com/Stacked-Org/stacked/assets/21108487/d1feb028-f8f9-4ee0-ae8e-5680c11a26bd)

I do believe this is the case with the other overlays as well (snackbars, bottomsheets).

### To reproduce

1. Create a dialog using `stacked create dialog test`
2. In the dialog file, add a type to the DialogRequest parameter 
3.  rebuild using `stacked generate`
4.  You will see an error in your `app.dialogs.dart` file. 

### Expected behavior

Instead of using dynamic type, it should cast it to the expected type upon generation.

![image](https://github.com/Stacked-Org/stacked/assets/21108487/c62cbb9b-55bb-4f2c-929d-797d0a271b97)

![image](https://github.com/Stacked-Org/stacked/assets/21108487/4b1117a7-f6a0-4a1f-be16-e84d371c0863)

**Comment by FilledStacks** [comment]
Just to check. 

A solution for this would be to add a the cast `request as DialogRequest<PAOfferInfo>` to the generated code?

**Comment by Chappie74** [comment]
Yes thats how I'm getting around it.  But of course, on regeneration it gets removed and have to be added again.

**Comment by FilledStacks** [comment]
I see.

We'll have to add that to the generator. Our time is currently fully on SessionMate but when we open up a slot we'll tackle this issue. 

Thanks for filing the request.
